test/integration/servicedns_test.go

```diff
@@ -204,3 +196,23 @@ func newServiceDNSTestFixture(tl common.TestLogger, fedFixture *framework.Federa
 
 	return f
 }
+
+func updateFederatedClusterStatus(tl common.TestLogger, client fedclientset.Interface, clusterName string, clusterStatus fedv1a1.FederatedClusterStatus) {
+	err := wait.PollImmediate(framework.DefaultWaitInterval, wait.ForeverTestTimeout, func() (exist bool, err error) {
+		federatedCluster, err := client.CoreV1alpha1().FederatedClusters(util.FederationSystemNamespace).Get(clusterName, metav1.GetOptions{})
+		if err != nil {
+			tl.Logf("Error retrieving federated cluster %q: %v", clusterName, err)
+			return false, err
+		}
+		federatedCluster.Status = clusterStatus
+		_, err = client.CoreV1alpha1().FederatedClusters(util.FederationSystemNamespace).UpdateStatus(federatedCluster)
+		if err != nil {
+			tl.Logf("Retry updating federated cluster status %q: %v", clusterName, err)
```

Comment:
`s/Retry/Error/`

Reply:
@marun, that's not error log. we will retry if update status failed.

Reply:
I was triggered by the inconsistency in handling of an error on retrieval of the federated cluster vs the update.  I had incorrectly assumed that returning an error would also retry, but having examined `poll` I see that is not the case.  Is there a reason retrieval vs update should have different retry behavior?

Reply:
retrieval of the object might fail due to various other reasons, like api server not reachable, etc... retrying in such cases will simply delay the failing test case. On the other hand status update (followed by object retrieval) failures happen mostly due to somebody else have updated it.